internal/cmd/oohelperd/internal/nwcth/http.go

```diff
@@ -0,0 +1,110 @@
+package nwcth
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"strings"
+
+	"github.com/ooni/probe-cli/v3/internal/iox"
+	"github.com/ooni/probe-cli/v3/internal/runtimex"
+)
+
+// HTTPConfig configures the HTTP check.
+type HTTPConfig struct {
+	Jar       *cookiejar.Jar
+	Headers   map[string][]string
+	Transport http.RoundTripper
+	URL       *url.URL
+}
+
+// HTTPDo performs the HTTP check.
+func HTTPDo(ctx context.Context, config *HTTPConfig) (*CtrlHTTPRequest, *NextLocationInfo) {
+	req, err := newRequest(ctx, config.URL)
+	if err != nil {
+		return &CtrlHTTPRequest{Failure: newfailure(err)}, nil
+	}
+	// The original test helper failed with extra headers while here
+	// we're implementing (for now?) a more liberal approach.
+	for k, vs := range config.Headers {
+		switch strings.ToLower(k) {
+		case "user-agent":
+		case "accept":
+		case "accept-language":
```

Comment:
Would you mind updating the specification (which currently is a comment in the reference issue) to bump it's version and to mention that those headers are the only ones that are passed along? (Also, once we do that, the comment can go away, since now we're following the _new_ spec.)

Reply:
I cannot edit the comment in the reference issue, it seems.

Reply:
oh, I didn't know, then this is a TODO(@bassosimone) :-)

Reply:
done here: https://github.com/ooni/spec/pull/219/commits/e7ec1e683bd4f1b158a41e632d9f705cd42e2a35